__tests__/app.test.js

```diff
@@ -334,3 +334,82 @@ describe('GET /api/reviews/:review_id/comments', () => {
     });
   });
 });
+
+describe('POST /api/reviews/:review_id/comments', () => {
+  test('201: responds with the posted comment', () => {
+    const postComment = {
+      username: 'dav3rid',
+      body: "My cats didn't want to know - they just scattered the pieces everywhere",
+    };
+    return request(app)
+      .post('/api/reviews/1/comments')
+      .send(postComment)
+      .expect(201)
+      .then(({ body: { postedComment } }) => {
+        expect(postedComment).toBe("My cats didn't want to know - they just scattered the pieces everywhere");
+      });
+  });
+  describe('errors', () => {
+    test('400: Bad Request response where :review_id is an invalid type', () => {
+      const postComment = {
+        username: 'dav3rid',
+        body: "My cats didn't want to know - they just scattered the pieces everywhere",
+      };
+      return request(app)
+        .post('/api/reviews/pizza/comments')
+        .send(postComment)
+        .expect(400)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe('Bad request');
+        });
+    });
+    test('404: Not found response where :review_id does not exist (but is valid type)', () => {
+      const postComment = {
+        username: 'dav3rid',
+        body: "My cats didn't want to know - they just scattered the pieces everywhere",
+      };
+      return request(app)
+        .post('/api/reviews/69/comments')
+        .send(postComment)
+        .expect(404)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe('Not found');
+        });
+    });
+    test('400: Bad request response where request body is missing information', () => {
+      const postComment = {
+        body: "My cats didn't want to know - they just scattered the pieces everywhere",
+      };
+      return request(app)
+        .post('/api/reviews/1/comments')
+        .send(postComment)
+        .expect(400)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe('Bad request');
+        });
+    });
+    test('400: Bad request response where request body is omitted', () => {
+      const postComment = {};
+      return request(app)
+        .post('/api/reviews/1/comments')
+        .send()
+        .expect(400)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe('Bad request');
+        });
+    });
+    test('400: Bad request response where username does not exist in users table', () => {
```

Comment:
I think this would fall under a 404 - as the user salehTheGreat could exist at some point :)